teos/cli/teos_cli.py

```diff
@@ -110,85 +155,267 @@ def stop(self):
         print("Closing the Eye of Satoshi")
 
 
-def main(command, args, data_dir, command_line_conf):
-    # Loads config and sets up the data folder and log file
-    config_loader = ConfigLoader(data_dir, CONF_FILE_NAME, DEFAULT_CONF, command_line_conf)
-    config = config_loader.build_config()
+class CliCommand:
+    """
+    Base class of each Cli command.
 
-    setup_data_folder(data_dir)
+    All the implementations should have an appropriately formatted docstring. See existing commands for an example.
+    Any implementation _must_ override the ``name`` attribute, and it might override the ``shortopts`` and ``longopts``
+    attributes.
+    """
 
-    teos_rpc_host = config.get("RPC_BIND")
-    teos_rpc_port = config.get("RPC_PORT")
+    name = None
```

Comment:
I think this variables could be part of `__init__`, and parse_args won't be a class method.

Reply:
Uhm, I think that would change the design. My idea here is that the subclasses of `CLICommand` are never instantiated, so there are only static or class methods. `parse_args` in `CLICommand` gives a reasonable default behavior (parsing with `getopt`, but subclasses might in principle override that with a completely different `parse_args`, which should nevertheless be stateless. This is also necessary if we want to register the commands with a decorator.

Reply:
Then you can at least remove `shortopts` and `longopts`, since they are never used. The latter is also setup to default for `getopt`, so doing:

```
return getopt(args, "")
```

Should do.